File: secondary_implementations/python/modules/RedditModule.py
# vim: ts=2 sw=2 noexpandtab
from __future__ import unicode_literals
import requests
from .AbstractModule import AbstractModule


class RedditModule(AbstractModule):
	def try_login(self, uname, password):
		payload = {
			'user': uname,
			'passwd': password,
			'api_type': 'json'}
		headers = {
			'user-agent': AbstractModule().user_agent(),
			'Content-Type': 'application/x-www-form-urlencoded'}
		loginResp = requests.post(
			'http://www.reddit.com/api/login/', data=payload, headers=headers)

		return 'wrong password' not in loginResp.text


reddit = RedditModule()

File: secondary_implementations/ruby/modules/RedditModule.rb
require_relative 'AbstractModule'  

class RedditModule < AbstractModule  
  def try_login(username, password)  
  	payload = {'user' => username, 'passwd' => password, 'api_type' => 'json'}
		headers = { 'user-agent' => user_agent, 'Content-Type' => 'application/x-www-form-urlencoded'}
		response = RestClient.post("http://www.reddit.com/api/login/", payload, :headers => headers){|response, request, result| response }
  	
  	return !(response.body.include? "wrong password")
  end  
end  